docs/cache-server.md

```diff
@@ -0,0 +1,130 @@
+
+### High-level overview
+
+The cache server is a regular, Kubernetes-style CRUD API server with support of LIST/WATCH semantics. 
+Conceptually it supports two modes of operations. 
+The first mode is in which a shard gets its private space for storing its own data. 
+The second mode is in which a shard can read other shards' data.
+
+The first mode of operation is implemented as a write controller that runs on a shard.
+It holds some state/data from that shard in memory and pushes it to the cache server.
+The controller uses standard informers for getting required resources both from a local kcp instance and remote cache server.
+Before pushing data it has to compare remote data to its local copy and make sure both copies are consistent.
+
+The second mode of operation is implemented as a read controller(s) that runs on some shard. 
+It holds other shards' data in an in-memory cache using an informer, 
+effectively pulling data from the central cache.
+Thanks to having a separate informer for interacting with the cache server 
+a shard can implement a different resiliency strategy.
+For example, it can tolerate the unavailability of the secondary during startup and become ready.
+
+### Running the server
+
+The cache server can be run as a standalone binary or as part of a kcp server.
+
+The standalone binary is in https://github.com/kcp-dev/kcp/tree/main/cmd/cache-server and can be run by issuing `go run ./cmd/cache-server/main.go` command.
+
+To run it as part of a kcp server, pass `--cache-url` flag to the kcp binary.
+
+### Client-side functionality
+
+In order to interact with the cache server from a shard, the https://github.com/kcp-dev/kcp/tree/main/pkg/cache/client 
+repository provides the following client-side functionality:
+
+[ShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L54), 
+a shard aware wrapper around `http.RoundTripper`. It changes the URL path to target a shard from the context.
+
+[DefaultShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L128) 
+is a `http.RoundTripper` that sets a default shard name if not specified in the context.
+
+For example, in order to make a client shard aware, inject the `http.RoundTrippers` to a `rest.Config`
+
+```go
+import (
+  cacheclient "github.com/kcp-dev/kcp/pkg/cache/client"
+  "github.com/kcp-dev/kcp/pkg/cache/client/shard"
+)
+
+// adds shards awareness to a client with a default `shard.Wildcard` name.
+NewClusterForConfig(cacheclient.WithShardRoundTripper(cacheclient.WithDefaultShardRoundTripper(serverConfig.LoopbackClientConfig, shard.Wildcard)))
+
+// and then change the context when you need to access a specific shard and pass is when making a HTTP request
+ctx = cacheclient.WithShardInContext(ctx, shard.New("cache"))
+```
+
+### Authorization/Authentication
+
+Not implemented at the moment
```

Comment:
do we have an issue for this?

Reply:
added a todo step to https://github.com/kcp-dev/kcp/issues/1225